python/ray/experimental/workflow/recovery.py

```diff
@@ -51,7 +52,13 @@ def _recover_workflow_step(input_object_refs: List[str],
     for index, _step_id in instant_workflow_inputs.items():
         # override input workflows with instant workflows
         input_workflows[index] = reader.load_step_output(_step_id)
-    input_object_refs = [reader.load_object_ref(r) for r in input_object_refs]
+
+    promises = []
```

Comment:
Could you confirm that this will be hidden into storage later (refactoring PR)? It exposes too much logic about converting input_object_refs. If yes, could you please add a TODO there?